test/e2e/snaps/test-snap-dialog.spec.js

```diff
@@ -0,0 +1,201 @@
+          tag: 'button',
+        });
+
+        // switch back to test snaps tab
+        windowHandles = await driver.waitUntilXWindowHandles(2, 1000, 10000);
+        await driver.switchToWindowWithTitle('Test Snaps', windowHandles);
+
+        // check result is null
+        result = await driver.findElement('#dialogResult');
+        await driver.delay(1000);
+        assert.equal(await result.getText(), 'null');
+
+        // click conf button
+        await driver.clickElement('#sendConfButton');
+        await driver.delay(1000);
+
+        // switch to dialog popup
+        windowHandles = await driver.waitUntilXWindowHandles(3, 1000, 10000);
+        windowHandles = await driver.getAllWindowHandles();
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+        await driver.delay(1000);
+
+        // click reject
+        await driver.clickElement({
+          text: 'Reject',
+          tag: 'button',
+        });
+
+        // switch back to test snaps tab
+        windowHandles = await driver.waitUntilXWindowHandles(2, 1000, 10000);
+        await driver.switchToWindowWithTitle('Test Snaps', windowHandles);
+
+        // check for false result
+        result = await driver.findElement('#dialogResult');
+        await driver.delay(1000);
+        assert.equal(await result.getText(), 'false');
+
+        // click conf button again
+        await driver.clickElement('#sendConfButton');
+        await driver.delay(1000);
+
+        // switch to dialog popup
+        windowHandles = await driver.waitUntilXWindowHandles(3, 1000, 10000);
+        windowHandles = await driver.getAllWindowHandles();
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+        await driver.delay(1000);
+
+        // click accept
+        await driver.clickElement({
+          text: 'Approve',
+          tag: 'button',
+        });
+
+        // switch back to test snaps tab
+        windowHandles = await driver.waitUntilXWindowHandles(2, 1000, 10000);
+        await driver.switchToWindowWithTitle('Test Snaps', windowHandles);
+
+        // check for true result
+        result = await driver.findElement('#dialogResult');
+        await driver.delay(1000);
+        assert.equal(await result.getText(), 'true');
+
+        // click prompt button
+        await driver.clickElement('#sendPromptButton');
+        await driver.delay(1000);
+
+        // switch to dialog popup
+        windowHandles = await driver.waitUntilXWindowHandles(3, 1000, 10000);
+        windowHandles = await driver.getAllWindowHandles();
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+        await driver.delay(1000);
+
+        // fill '2323' in form field
+        await driver.fill('.MuiInput-input', '2323');
+
+        // click submit button
+        await driver.clickElement({
```

Comment:
Should we test the reject case for the prompt as well?

Reply:
I've added this in the latest commit

Reply:
Test